src/runtime/runtime_unix.go

```diff
@@ -217,8 +222,22 @@ func nanosecondsToTicks(ns int64) timeUnit {
 }
 
 func sleepTicks(d timeUnit) {
+	// Check for incoming signals.
+	if checkSignals() {
+		// Received a signal, so there's probably at least one goroutine that's
+		// runnable again.
+		return
+	}
+
+	// TODO: there is a race condition here. If a signal arrives between
+	// checkSignals() and usleep(), the usleep() call will not exit early so the
+	// signal is delayed until usleep finishes or another signal arrives.
```

Comment:
Iirc we talked about this issue and the only way to get this working asynchronously is by adding a second listening thread/go routine which is not possible at the moment, right? So for the initial support, we should not see this as a blocking factor.